### ISSUE TITLE
CudaFloatType is at version 3 and the expected version is 2.

### ISSUE THREAD

**Issue by JVedant** [pytorch/pytorch#59706]
## 🐛 Bug

When I try to train a Adversarial neural network with the nightly build of Pytorch, I get the error saying "CUDAFloatType [3]] is at version 3; expected version 2 instead"

## To Reproduce

1. Training any neural network with the below mentioned configuration throws the same error

`terminate called after throwing an instance of 'std::runtime_error'
what():  one of the variables needed for gradient computation has been modified by an inplace operation: [CUDAFloatType [3]] is at version 3; expected version 2 instead. Hint: the backtrace further above shows the operation that failed to compute its gradient. The variable in question was changed in there or anywhere later. Good luck!
Aborted (core dumped)`

## Expected behavior
The neural network should train

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): '1.9.0.dev20210601+cu113'
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version: Cuda: 11.2
 - GPU models and configuration: 1x RTX 3090 (24 Gigs)
 - Any other relevant information:

## Additional context

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @Varal7 @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @mruberry @ngimel

**Comment by ngimel** [comment]
The error message is saying that you modified a variable needed for the autograd, that's most likely a user error. If you think that it's not, please provide a reproducible example.

**Comment by JVedant** [comment]
Thanks for the reply, I changed the optimizer from Adam to RMSProp and it worked as expected. (Not sure where the problem was)
